test/react/FirstRenderOnly/equivalence3.js

```diff
@@ -1,27 +1,27 @@
 var React = require("react");
 
 function App(props) {
-  var switch = props.switch;
+  var foo = props.foo;
   var x = {
     val: 1,
   };
   var y = {
     val: 1,
-  }
-  var foo = Object.assign({}, props, switch ? x : y);
+  };
+  var foo = Object.assign({}, props, foo ? x : y);
   var a = <span {...foo} key={null} />;
   var b = <span {...foo} key={null} />;
   return [a, b];
 }
 
 App.getTrials = function(renderer, Root, data, isCompiled) {
   if (isCompiled) {
-    const [a, b] = App({});
+    const [a, b] = App({foo: false});
```

Comment:
Wait. This looks confusing.

We shouldn't ever be using `App` in this test directly I think. It always led to bugs in the past. It should be using `Root` instead.

Was this intentional?